### PR TITLE
DynamoDBToS3Operator - Add a feature to export the table to a point in time.

### LINKED ISSUE #28830: Export DynamoDB table to S3 with PITR
### Description

Airflow provides the Amazon DynamoDB to Amazon S3 below.  
https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/transfer/dynamodb_to_s3.html

Most of Data Engineer build their "export DDB data to s3" pipeline using "within the point in time recovery window".   
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.export_table_to_point_in_time

I appreciate if airflow has this function as a native function.  

### Use case/motivation

My daily batch job exports its data with pitr option. All of tasks is written by apache-airflow-providers-amazon except "export_table_to_point_in_time" task. 
"export_table_to_point_in_time" task only used the python operator. I expect I can unify the task as apache-airflow-providers-amazon library. 

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by utkarsharma2**
DynamoDBToS3Operator - Add a feature to export the table to a point in time. 
Fixed - Now fetching the TableARN from `client.describe_table(TableName=self.dynamodb_table_name)` and added `file_size` in the example.

closes: https://github.com/apache/airflow/issues/28830

### CODE REVIEW

File: airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py [hunk 1]
```diff
@@ -118,11 +126,44 @@ def __init__(
...
+        else:
+            self._export_entire_data()
+
+    def _export_table_to_point_in_time(self):
+        """
+        Export data from start of epoc till `export_time`. Table export will be a snapshot of the table's
+         state at this point in time.
+        """
+        client = self.hook.conn.meta.client
+        table_description = client.describe_table(TableName=self.dynamodb_table_name)
```
Comment (utkarsharma2) [inline airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py:152]: Fixed TableARN issue

File: tests/system/providers/amazon/aws/example_dynamodb_to_s3.py [hunk 2]
```diff
@@ -127,6 +127,17 @@ def delete_dynamodb_table(table_name: str):
         },
     )
     # [END howto_transfer_dynamodb_to_s3_segmented]
+
+    # [START howto_transfer_dynamodb_to_s3_in_some_point_in_time]
+    backup_db_to_point_in_time = DynamoDBToS3Operator(
+        task_id="backup_db_to_point_in_time",
+        dynamodb_table_name=table_name,
+        file_size=1000,
```
Comment (utkarsharma2) [inline tests/system/providers/amazon/aws/example_dynamodb_to_s3.py:135]: Added required param.

**Comment by utkarsharma2** [comment]
Have tested this locally with the below dag and it's working for me
```
from datetime import datetime

from airflow import DAG
from airflow.providers.amazon.aws.transfers.dynamodb_to_s3 import DynamoDBToS3Operator

with DAG(
    dag_id='example_export_dynamodb',
    schedule_interval=None,
    start_date=datetime(2021, 1, 1),
    tags=['example'],
    catchup=False,
) as dag:
    dynamodb_to_s3_operator = DynamoDBToS3Operator(
        task_id="dynamodb_to_s3",
        dynamodb_table_name="test",
        s3_bucket_name="tmp9",
        file_size=4000,
        export_time=datetime.now(),
        aws_conn_id="aws_default",
        s3_key_prefix="test1"
    )
 ```

Screenshot 2023-05-09 at 11 50 41 AM

File: airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py [hunk 3]
```diff
@@ -118,11 +126,44 @@ def __init__(
...
+        response = client.export_table_to_point_in_time(
+            TableArn=table_description.get("Table", {}).get("TableArn"),
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py:154]: I saw your question in the other PR before I saw you already submitted this one.  Glad you came tot his solution anyway :)    As I mentioned over there, another possible solution would have been to use cached_property to get and store the ARN, but I don't think there is much difference in this case in the end.  :+1:

**Review by ferruzzi — APPROVED** [review]
Thanks for not only taking on the fixes, but getting the done so quick.   LGTM (again :facepalm:)

**Review by vincbeck — APPROVED** [review]

File: airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py (hunk 3 above)
Comment (vincbeck) [reply airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py:154]: +1 on this one :) Thanks for your reactivity!

**Review by o-nikolas — APPROVED** [review]
Thanks for the quick turnaround!

File: airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py (hunk 3 above)
Comment (utkarsharma2) [reply airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py:154]: Thanks for a quick review. I'll be more mindful going forward with respect to system tests. 👍

**Comment by ferruzzi** [comment]
@utkarsharma2    System tests still failing with the mesage 

```


Traceback (most recent call last):
--
File "/opt/airflow/airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py", line 142, in execute
self._export_table_to_point_in_time()
File "/opt/airflow/airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py", line 158, in _export_table_to_point_in_time
ExportFormat=self.export_format,
File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 530, in _api_call
return self._make_api_call(operation_name, kwargs)
File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 960, in _make_api_call
raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the ExportTableToPointInTime operation: Invalid Request: 1 validation error detected: Value '' at 's3Prefix' failed to satisfy constraint: Member must have length greater than or equal to 3

```

**Comment by ferruzzi** [comment]
I'm at Open source Summit so I can't look too closely right now, but I wonder if the prefix has to start with `s3:` or `s3/` or something like that?

**Comment by ferruzzi** [comment]
Or if a blank prefix is viable, maybe try using airflow.helpers.prune_dict() to drop the None value

**Comment by utkarsharma2** [comment]
@ferruzzi I was able to reproduce the issue on local and there was a missing key `s3_key_prefix` which I have added and the format for export_time was not correct both of the issues have been fixed in PR - https://github.com/apache/airflow/pull/31197 and I have tested and verified below code on my local. 
```
backup_db_to_point_in_time = DynamoDBToS3Operator(
        task_id="backup_db_to_point_in_time",
        dynamodb_table_name=table_name,
        file_size=1000,
        s3_bucket_name=bucket_name,
        export_time=datetime.now(),
        s3_key_prefix=f"{S3_KEY_PREFIX}-3-",
    )
```

**Pull request merged**